src/network/wait-online/manager.c

```diff
@@ -60,7 +67,31 @@ static int manager_link_is_online(Manager *m, Link *l, LinkOperationalStateRange
                 return 0;
         }
 
+        required_family = m->required_family > 0 ? m->required_family : l->required_family;
+        needs_ipv4 = required_family & ADDRESS_FAMILY_IPV4;
+        needs_ipv6 = required_family & ADDRESS_FAMILY_IPV6;
+
+        if (s.min >= LINK_OPERSTATE_DEGRADED) {
+                if (needs_ipv4 && l->ipv4_address_state < LINK_ADDRESS_STATE_DEGRADED)
+                        goto family_not_ready;
+
+                if (needs_ipv6 && l->ipv6_address_state < LINK_ADDRESS_STATE_DEGRADED)
+                        goto family_not_ready;
+        }
+
+        if (s.min >= LINK_OPERSTATE_ROUTABLE) {
+                if (needs_ipv4 && l->ipv4_address_state < LINK_ADDRESS_STATE_ROUTABLE)
+                        goto family_not_ready;
+
+                if (needs_ipv6 && l->ipv6_address_state < LINK_ADDRESS_STATE_ROUTABLE)
+                        goto family_not_ready;
+        }
+
         return 1;
+
+family_not_ready:
+        log_link_debug(l, "Required address family '%s' is not reached", link_required_address_family_to_string(required_family));
```

Comment:
`address family is not reached` sounds slightly strange for me. How about the following: 
```
if (s.min >= LINK_OPERSTATE_DEGRADED) {
        if (needs_ipv4 && l->ipv4_address_state < LINK_ADDRESS_STATE_DEGRADED) {
                log_link_debug(l, "No routable or link-local IPv4 address is configured.");
                return 0;
        }
        ...
}
```